File: core/src/main/scala/com/graphbrain/db/ProgNode.scala
package com.graphbrain.db

case class ProgNode(override val id: String,
                    prog: String="",
                    override val degree: Int = 0,
                    override val ts: Long = -1)
  extends Vertex (id, degree, ts) {

  def this(id: String, map: Map[String, String]) =
    this(id,
      map("prog"),
      map("degree").toInt,
      map("ts").toLong)

  override def extraMap = Map("prog" -> prog)

  override def setId(newId: String): Vertex = copy(id=newId)

  override def setDegree(newDegree: Int): Vertex = copy(degree=newDegree)

  override def setTs(newTs: Long): Vertex = copy(ts=newTs)
}

File: core/src/main/java/com/graphbrain/db/ProgNode.java
package com.graphbrain.db;

import java.util.Map;

public class ProgNode extends Vertex {

    private String prog;

    @Override
    public VertexType type() {return VertexType.Prog;}

    public ProgNode(String id, String prog, int degree, long ts) {
        super(id, degree, ts);
        this.prog = prog;
    }

    public ProgNode(String id, String prog) {
        this(id, prog, 0, -1);
    }

    @Override
    public Vertex copy() {
        return new ProgNode(id, prog, degree, ts);
    }

    public String getProg() {
        return prog;
    }
}